pkg/sfu/streamtracker/streamtracker.go

```diff
@@ -18,31 +20,24 @@ func (s StreamStatus) String() string {
 	case StreamStatusActive:
 		return "active"
 	default:
-		return "unknown"
+		return fmt.Sprintf("unknown: %d", int(s))
 	}
 }
 
 const (
-	StreamStatusStopped StreamStatus = 0
-	StreamStatusActive  StreamStatus = 1
+	StreamStatusStopped StreamStatus = iota
+	StreamStatusActive
 )
 
-type StreamTrackerParams struct {
-	// number of samples needed per cycle
-	SamplesRequired uint32
-
-	// number of cycles needed to be active
-	CyclesRequired uint32
-
-	CycleDuration time.Duration
+// ------------------------------------------------------------
 
+type StreamTrackerParams struct {
+	StreamTrackerImpl     StreamTrackerImpl
```

Comment:
This is the `StreamTracker` base struct that should be passed in an impl. This class does things like measuring bitrate, handling pause state, stream state etc.

I did another design where the actual implementation is the top level struct and those include this base struct so that callers could instantiate an implementation and be done with it. Now callers have to instantiate an impl and then put that in the params of base and instantiate base. Not too bad, but not ideal. But, the base embedded in top level became too complicate with locking and code sharing. So, went with this approach which turned out to be cleaner.